providers/ofrep/src/main/java/dev/openfeature/contrib/providers/ofrep/internal/OfrepApi.java

```diff
@@ -0,0 +1,131 @@
+package dev.openfeature.contrib.providers.ofrep.internal;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import java.io.IOException;
+import java.net.ProxySelector;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.Executor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * The OfrepApi class is responsible for communicating with the OFREP server to evaluate flags.
+ */
+@Slf4j
+public class OfrepApi {
+
+    private static final String path = "/ofrep/v1/evaluate/flags/";
+    private final ObjectMapper serializer;
+    private final ObjectMapper deserializer;
+    private final HttpClient httpClient;
+    private final Duration requestTimeout;
+    private Instant nextAllowedRequestTime = Instant.now();
+
+    /**
+     * Constructs an OfrepApi instance with a HTTP client and JSON serializers.
+     *
+     * @param requestTimeout - The request timeout duration for the request.
+     * @param connectTimeout - The connect timeout duration for establishing HTTP connection.
+     * @param proxySelector  - The ProxySelector to use for HTTP requests.
+     * @param executor       - The Executor to use for operations.
+     */
+    public OfrepApi(Duration requestTimeout, Duration connectTimeout, ProxySelector proxySelector, Executor executor) {
+        httpClient = HttpClient.newBuilder()
+                .connectTimeout(connectTimeout)
```

Comment:
We could possibly add keepAlive settings, but by default I think there's a 30s pooled connection so I think that's good for now.